### PR TITLE
Good morning, NTSS Dagon! AKA Adds Landmines

### PR DESCRIPTION
**Pull request by ProbablyCarl**
- - -
Additions:
 - Ripped straight from Virgo, item versions of landmines. Two have been added to the Traitor uplink.
 - For mappers, a random loot spawn for landmines, containing each type in the effect state, has been added.
 - Radio Host. Given an area south of deck three EVA, they're probably going to be nothing more than a space gremlin on meth with access to a radio. We'll see.
- - -
Other:
 - Multiple off-sites given 'semi-antag' status. This is in line with a change Gabo had lined up.

### CODE REVIEW

**Comment by chaoko99** [comment]
>Radio Host. Given an area south of deck three EVA, they're probably going to be nothing more than a space gremlin on meth with access to a radio. We'll see.

Don't we literally have an actor role for this exact purpose?

File: code/game/objects/effects/mines.dm [hunk 1]
```diff
@@ -103,3 +103,61 @@
...
+	return
+
+/obj/item/weapon/mine/proc/prime(mob/user as mob, var/explode_now = FALSE)
+	visible_message("\The [src.name] beeps as the priming sequence completes.")
+	var/obj/effect/mine/R = new minetype(get_turf(src))
+	src.transfer_fingerprints_to(R)
+	R.add_fingerprint(user)
+	if(explode_now)
+		R.explode(user)
+	spawn(0)
```
Comment (chaoko99) [inline code/game/objects/effects/mines.dm:137]: `QDEL_IN(src, 0)`

**Comment by ProbablyCarl** [comment]
> > Radio Host. Given an area south of deck three EVA, they're probably going to be nothing more than a space gremlin on meth with access to a radio. We'll see.
> 
> Don't we literally have an actor role for this exact purpose?

Not on station, no. The point of the role on station is to promote crew interaction, among other things. We've had it test merged, and it's quite a laugh at times. Besides, no one plays actor. Like, at all.

**Comment by Roland410** [comment]
I'd still say that radio host should be a civvie/contractor role, idk how it would fit in for NTEF/SMC.

**Comment by Roland410** [comment]
Also, rename the radio host's stuff in code from 'radio' to 'radiohost' or something, as it'll be confusing as fuck in the future.

File: code/game/objects/effects/mines.dm [hunk 2]
```diff
@@ -103,3 +103,61 @@
 	name = "Stun Mine"
 	icon_state = "uglymine"
 	triggerproc = "triggerstun"
+
```
Comment (chaoko99) [inline code/game/objects/effects/mines.dm:106]: Couldn't like, all of these just be items with crossed events? Why are they items that spawn mines? Anchoring an object makes it impossible to pick up.

**Comment by Roland410** [comment]
Radio host mapping makes trash end up in their room apparently, and if they throw it in the disposals, it goes into tcomms. Travis also agrees on that.

File: code/game/objects/effects/mines.dm (hunk 1 above)
Comment (BurgerLUA) [inline code/game/objects/effects/mines.dm:137]: Add the QDEL_IN(src,0) here and test if it still works as intended.

**Comment by ProbablyCarl** [comment]
I've no clue why the test is failing. I'm a bit tired as of current, which is probably why, but I can't see anything that should be causing the failure.

**Comment by ProbablyCarl** [comment]
The disposal issue, among other requested changes, have been fixed.
It's just this Travis issue now, I think.

File: maps/torch/items/cards_ids.dm [hunk 3]
```diff
@@ -228,4 +228,7 @@
 
 /obj/item/weapon/card/id/torch/crew/psiadvisor
 	job_access_type = /datum/job/psiadvisor
-	detail_color = COLOR_COMMAND_BLUE
\ No newline at end of file
+	detail_color = COLOR_COMMAND_BLUE
+
+/obj/item/weapon/card/id/torch/crew/radio
+	job_access_type = /datum/job/radio
```
Comment (Roland410) [inline maps/torch/items/cards_ids.dm:234]:
```suggestion
/obj/item/weapon/card/id/torch/crew/radio_host
	job_access_type = /datum/job/radio_host
```

File: maps/torch/job/hestia_jobs.dm [hunk 4]
```diff
@@ -217,3 +217,36 @@
 /datum/job/psiadvisor/get_description_blurb()
 	return "You are the Psionic Advisor, an agent of either the Foundation or Nanotrasen Psionic Corps. Alongside the Counselor, you're the only other individual with known and authorized Psionic abilities aboard the NTSS Dagon. Your main responsibility is advising the Commanding Officer on psionic matters. \
 	Secondly, you're to assist the crew or Research on psionic matters, or guide any newly emergent crew that awaken with psionic abilities."
+
+/datum/job/radio
```
Comment (Roland410) [inline maps/torch/job/hestia_jobs.dm:221]:
```suggestion
/datum/job/radio_host
```

File: maps/torch/job/hestia_jobs.dm [hunk 5]
```diff
@@ -217,3 +217,36 @@
...
+	title = "Radio Host"
+	department = "Service"
+	department_flag = SRV
+	total_positions = 1
+	spawn_positions = 1
+	minimal_player_age = 2
+	supervisors = "your journalistic morals"
+	selection_color = "#61a82f"
+	minimum_character_age = list(SPECIES_HUMAN = 18)
+	outfit_type = /decl/hierarchy/outfit/job/torch/radio
```
Comment (Roland410) [inline maps/torch/job/hestia_jobs.dm:231]:
```suggestion
	outfit_type = /decl/hierarchy/outfit/job/torch/radio_host
```

File: maps/torch/job/hestia_jobs.dm [hunk 6]
```diff
@@ -217,3 +217,36 @@
...
+		/datum/mil_rank/marine_corps/e3
+	)
+	access = list(access_radio_host, access_maint_tunnels, access_external_airlocks,
+				access_eva, access_RC_announce, access_solgov_crew, access_hangar)
+	minimal_access = list()
+	alt_titles = list(
+		"Spokesman",
+		"Artist")
+
+/datum/job/radio/get_description_blurb()
```
Comment (Roland410) [inline maps/torch/job/hestia_jobs.dm:251]:
```suggestion
/datum/job/radio_host/get_description_blurb()
```

File: maps/torch/job/outfits/boh_outfits.dm [hunk 7]
```diff
@@ -177,4 +177,11 @@
...
-	H.equip_to_slot_or_store_or_drop(regis_card)
\ No newline at end of file
+	H.equip_to_slot_or_store_or_drop(regis_card)
+
+//radio host
+/decl/hierarchy/outfit/job/torch/radio
+	name = OUTFIT_JOB_NAME("Radio Host")
+	uniform = /obj/item/clothing/under/sl_suit
+	shoes = /obj/item/clothing/shoes/laceup
+	id_type = /obj/item/weapon/card/id/torch/crew/radio
```
Comment (Roland410) [inline maps/torch/job/outfits/boh_outfits.dm:187]:
```suggestion
//radio host
/decl/hierarchy/outfit/job/torch/radio_host
	name = OUTFIT_JOB_NAME("Radio Host")
	uniform = /obj/item/clothing/under/sl_suit
	shoes = /obj/item/clothing/shoes/laceup
	id_type = /obj/item/weapon/card/id/torch/crew/radio_host
```

File: maps/torch/job/torch_jobs.dm [hunk 8]
```diff
@@ -40,7 +40,7 @@
 						/datum/job/janitor, /datum/job/chef, /datum/job/bartender,
 						/datum/job/senior_scientist, /datum/job/scientist, /datum/job/scientist_assistant,
 						/datum/job/ai, /datum/job/cyborg,
-						/datum/job/crew, /datum/job/assistant,
+						/datum/job/crew, /datum/job/assistant, /datum/job/radio,
```
Comment (Roland410) [inline maps/torch/job/torch_jobs.dm:43]:
```suggestion
						/datum/job/crew, /datum/job/assistant, /datum/job/radio_host,
```

File: maps/torch/loadout/_defines.dm [hunk 9]
```diff
@@ -1,16 +1,16 @@
...
 
 //For jobs that allow for decorative or ceremonial clothing
-#define FORMAL_ROLES list(/datum/job/liaison, /datum/job/bodyguard, /datum/job/rd, /datum/job/senior_scientist, /datum/job/scientist, /datum/job/scientist_assistant, /datum/job/psychiatrist, /datum/job/representative, /datum/job/assistant, /datum/job/bartender, /datum/job/merchant, /datum/job/detective, /datum/job/chaplain, /datum/job/psiadvisor, /datum/job/submap/bearcat_captain, /datum/job/submap/bearcat_crewman, /datum/job/submap/unishi_crew, /datum/job/submap/unishi_researcher, /datum/job/submap/colonist, /datum/job/submap/pod)
+#define FORMAL_ROLES list(/datum/job/liaison, /datum/job/bodyguard, /datum/job/rd, /datum/job/senior_scientist, /datum/job/scientist, /datum/job/scientist_assistant, /datum/job/psychiatrist, /datum/job/representative, /datum/job/assistant, /datum/job/bartender, /datum/job/merchant, /datum/job/detective, /datum/job/chaplain, /datum/job/psiadvisor, /datum/job/submap/bearcat_captain, /datum/job/submap/bearcat_crewman, /datum/job/submap/unishi_crew, /datum/job/submap/unishi_researcher, /datum/job/submap/colonist, /datum/job/submap/pod, /datum/job/radio)
```
Comment (Roland410) [inline maps/torch/loadout/_defines.dm:4]:
```suggestion
#define FORMAL_ROLES list(/datum/job/liaison, /datum/job/bodyguard, /datum/job/rd, /datum/job/senior_scientist, /datum/job/scientist, /datum/job/scientist_assistant, /datum/job/psychiatrist, /datum/job/representative, /datum/job/assistant, /datum/job/bartender, /datum/job/merchant, /datum/job/detective, /datum/job/chaplain, /datum/job/psiadvisor, /datum/job/submap/bearcat_captain, /datum/job/submap/bearcat_crewman, /datum/job/submap/unishi_crew, /datum/job/submap/unishi_researcher, /datum/job/submap/colonist, /datum/job/submap/pod, /datum/job/radio_host)
```

File: maps/torch/loadout/_defines.dm [hunk 10]
```diff
@@ -1,16 +1,16 @@
...
 
 //For civilian jobs that may have a uniform, but not a strict one
-#define SEMIFORMAL_ROLES list(/datum/job/assistant, /datum/job/mining, /datum/job/scientist_assistant, /datum/job/psychiatrist, /datum/job/bartender, /datum/job/merchant, /datum/job/nt_pilot, /datum/job/scientist, /datum/job/senior_scientist, /datum/job/detective, /datum/job/chaplain, /datum/job/roboticist, /datum/job/psiadvisor, /datum/job/submap/bearcat_captain, /datum/job/submap/bearcat_crewman, /datum/job/submap/unishi_crew, /datum/job/submap/unishi_researcher, /datum/job/submap/colonist, /datum/job/submap/pod)
+#define SEMIFORMAL_ROLES list(/datum/job/assistant, /datum/job/mining, /datum/job/scientist_assistant, /datum/job/psychiatrist, /datum/job/bartender, /datum/job/merchant, /datum/job/nt_pilot, /datum/job/scientist, /datum/job/senior_scientist, /datum/job/detective, /datum/job/chaplain, /datum/job/roboticist, /datum/job/psiadvisor, /datum/job/submap/bearcat_captain, /datum/job/submap/bearcat_crewman, /datum/job/submap/unishi_crew, /datum/job/submap/unishi_researcher, /datum/job/submap/colonist, /datum/job/submap/pod, /datum/job/radio)
```
Comment (Roland410) [inline maps/torch/loadout/_defines.dm:7]:
```suggestion
#define SEMIFORMAL_ROLES list(/datum/job/assistant, /datum/job/mining, /datum/job/scientist_assistant, /datum/job/psychiatrist, /datum/job/bartender, /datum/job/merchant, /datum/job/nt_pilot, /datum/job/scientist, /datum/job/senior_scientist, /datum/job/detective, /datum/job/chaplain, /datum/job/roboticist, /datum/job/psiadvisor, /datum/job/submap/bearcat_captain, /datum/job/submap/bearcat_crewman, /datum/job/submap/unishi_crew, /datum/job/submap/unishi_researcher, /datum/job/submap/colonist, /datum/job/submap/pod, /datum/job/radio_host)
```

File: maps/torch/loadout/_defines.dm [hunk 11]
```diff
@@ -1,16 +1,16 @@
...
-#define SEMIANDFORMAL_ROLES list(/datum/job/assistant, /datum/job/mining, /datum/job/scientist_assistant, /datum/job/psychiatrist, /datum/job/bartender, /datum/job/merchant, /datum/job/nt_pilot, /datum/job/liaison, /datum/job/bodyguard, /datum/job/rd, /datum/job/senior_scientist, /datum/job/scientist, /datum/job/representative, /datum/job/detective, /datum/job/chaplain, /datum/job/psiadvisor, /datum/job/submap/bearcat_captain, /datum/job/submap/bearcat_crewman, /datum/job/submap/unishi_crew, /datum/job/submap/unishi_researcher, /datum/job/submap/colonist, /datum/job/submap/pod)
+#define SEMIANDFORMAL_ROLES list(/datum/job/assistant, /datum/job/mining, /datum/job/scientist_assistant, /datum/job/psychiatrist, /datum/job/bartender, /datum/job/merchant, /datum/job/nt_pilot, /datum/job/liaison, /datum/job/bodyguard, /datum/job/rd, /datum/job/senior_scientist, /datum/job/scientist, /datum/job/representative, /datum/job/detective, /datum/job/chaplain, /datum/job/psiadvisor, /datum/job/submap/bearcat_captain, /datum/job/submap/bearcat_crewman, /datum/job/submap/unishi_crew, /datum/job/submap/unishi_researcher, /datum/job/submap/colonist, /datum/job/submap/pod, /datum/job/radio)
```
Comment (Roland410) [inline maps/torch/loadout/_defines.dm:10]:
```suggestion
#define SEMIANDFORMAL_ROLES list(/datum/job/assistant, /datum/job/mining, /datum/job/scientist_assistant, /datum/job/psychiatrist, /datum/job/bartender, /datum/job/merchant, /datum/job/nt_pilot, /datum/job/liaison, /datum/job/bodyguard, /datum/job/rd, /datum/job/senior_scientist, /datum/job/scientist, /datum/job/representative, /datum/job/detective, /datum/job/chaplain, /datum/job/psiadvisor, /datum/job/submap/bearcat_captain, /datum/job/submap/bearcat_crewman, /datum/job/submap/unishi_crew, /datum/job/submap/unishi_researcher, /datum/job/submap/colonist, /datum/job/submap/pod, /datum/job/radio_host)
```

File: maps/torch/loadout/_defines.dm [hunk 12]
```diff
@@ -1,16 +1,16 @@
...
 
 //For civilian jobs with no uniform or formal clothing requirements.
-#define CASUAL_ROLES list(/datum/job/assistant, /datum/job/janitor, /datum/job/chef, /datum/job/bartender, /datum/job/cargo_tech, /datum/job/roboticist, /datum/job/mining, /datum/job/chaplain, /datum/job/merchant, /datum/job/psiadvisor, /datum/job/submap/bearcat_captain, /datum/job/submap/bearcat_crewman, /datum/job/submap/unishi_crew, /datum/job/submap/unishi_researcher, /datum/job/submap/colonist, /datum/job/submap/pod)
+#define CASUAL_ROLES list(/datum/job/assistant, /datum/job/janitor, /datum/job/chef, /datum/job/bartender, /datum/job/cargo_tech, /datum/job/roboticist, /datum/job/mining, /datum/job/chaplain, /datum/job/merchant, /datum/job/psiadvisor, /datum/job/submap/bearcat_captain, /datum/job/submap/bearcat_crewman, /datum/job/submap/unishi_crew, /datum/job/submap/unishi_researcher, /datum/job/submap/colonist, /datum/job/submap/pod, /datum/job/radio)
```
Comment (Roland410) [inline maps/torch/loadout/_defines.dm:13]:
```suggestion
#define CASUAL_ROLES list(/datum/job/assistant, /datum/job/janitor, /datum/job/chef, /datum/job/bartender, /datum/job/cargo_tech, /datum/job/roboticist, /datum/job/mining, /datum/job/chaplain, /datum/job/merchant, /datum/job/psiadvisor, /datum/job/submap/bearcat_captain, /datum/job/submap/bearcat_crewman, /datum/job/submap/unishi_crew, /datum/job/submap/unishi_researcher, /datum/job/submap/colonist, /datum/job/submap/pod, /datum/job/radio_host)
```

File: maps/torch/torch_areas.dm [hunk 13]
```diff
@@ -1756,4 +1756,20 @@
...
+	icon_state = "chapel"
+	req_access = list(access_radio_host)
+
+/area/radio/interior
+	name = "\improper Studio"
+	icon_state = "heads"
+
+/area/radio/backroom
+	name = "\improper Studio Backroom"
+	icon_state = "Sleep"
```
Comment (Roland410) [inline maps/torch/torch_areas.dm:1775]:
```suggestion
/////////
// Radio Host Studio
/////////
/area/radio_host
	name = "\improper Studio Lobby"
	icon_state = "chapel"
	req_access = list(access_radio_host)

/area/radio_host/interior
	name = "\improper Studio"
	icon_state = "heads"

/area/radio_host/backroom
	name = "\improper Studio Backroom"
	icon_state = "Sleep"
```

**Comment by BurgerLUA** [comment]
One of your areas had incorrect power values. I don't know why this atom isn't atomized. I legitimately thought this PR added only landmines but for some reason you're adding an entirely new job that should be seperate from landmines.

**Comment by ProbablyCarl** [comment]
Gonna close this.
I'll atomize the PR and open two separate ones when I get the chance.

**Pull request closed without merging**